src/types/zodTypes.ts

```diff
@@ -0,0 +1,172 @@
+import { z } from 'zod';
+import { isDefinedAndNotNullAndNotEmpty } from '@rudderstack/integrations-lib';
+import { isHttpStatusSuccess } from '../v0/util';
+
+export const ProxyMetadataSchema = z.object({
+  jobId: z.number(),
+  attemptNum: z.number(),
+  userId: z.string(),
+  sourceId: z.string(),
+  destinationId: z.string(),
+  workspaceId: z.string(),
+  secret: z.record(z.unknown()),
```

Comment:
https://github.com/rudderlabs/rudder-server/blob/master/router/transformer/transformer_proxy_adapter.go#L26

https://github.com/rudderlabs/rudder-server/blob/master/router/transformer/transformer.go#L67

From server it will be sent. Possible it could be empty

Reply:
can we make it optional